addons/sale_timesheet/models/sale_order.py

```diff
@@ -126,41 +127,61 @@ class SaleOrderLine(models.Model):
-                        remaining_time = ' ({sign}{hours:02.0f}:{minutes:02.0f})'.format(
-                            sign='-' if line.remaining_hours < 0 else '',
-                            hours=hours,
-                            minutes=minutes)
-                    elif encoding_uom == uom_day:
-                        remaining_days = company.project_time_mode_id._compute_quantity(line.remaining_hours, encoding_uom, round=False)
-                        remaining_time = ' ({qty:.02f} {unit})'.format(
-                            qty=remaining_days,
-                            unit=_('days') if abs(remaining_days) > 1 else _('day')
+                if with_remaining_hours:
+                    name = names.get(line.id)
+                    if line.remaining_hours_available:
+                        company = self.env.company
+                        encoding_uom = company.timesheet_encode_uom_id
+                        remaining_time = ''
+                        if encoding_uom == uom_hour:
+                            hours, minutes = divmod(abs(line.remaining_hours) * 60, 60)
+                            round_minutes = minutes / 30
+                            minutes = math.ceil(round_minutes) if line.remaining_hours >= 0 else math.floor(round_minutes)
+                            if minutes > 1:
+                                minutes = 0
+                                hours += 1
+                            else:
+                                minutes = minutes * 30
+                            remaining_time = ' ({sign}{hours:02.0f}:{minutes:02.0f})'.format(
+                                sign='-' if line.remaining_hours < 0 else '',
+                                hours=hours,
+                                minutes=minutes)
+                        elif encoding_uom == uom_day:
+                            remaining_days = company.project_time_mode_id._compute_quantity(line.remaining_hours, encoding_uom, round=False)
+                            remaining_time = ' ({qty:.02f} {unit})'.format(
+                                qty=remaining_days,
+                                unit=_('days') if abs(remaining_days) > 1 else _('day')
+                            )
+                        name = '{name}{remaining_time}'.format(
+                            name=name,
+                            remaining_time=remaining_time
                         )
-                    name = '{name}{remaining_time}'.format(
-                        name=name,
-                        remaining_time=remaining_time
-                    )
-                result.append((line.id, name))
+                        if with_price_unit:
+                            names[line.id] = name
+                    if not with_price_unit:
+                        result.append((line.id, name))
+                if with_price_unit:
+                    sols_by_so_dict[line.order_id.id, line.product_id.id] += line
+
+            if with_price_unit:
```

Comment:
Can't we integrate this process directly in the for `line in self:` loop?

Reply:
Perhaps, in fact, I need to know if there are many sales order items with a same product in a sale order. If yes then we can add the price_unit in the name_get of these sales order items. I don't see how can I do directly in the foreach loop

Reply:
Ok my bad :) I did not get that we needed to display the information only when we had several lines of the same product with different prises.